lang/csharp/src/apache/main/Reflect/ClassCache.cs

```diff
@@ -0,0 +1,209 @@
+/*  Copyright 2019 Pitney Bowes Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Reflection;
+using Avro;
+
+namespace Avro.Reflect
+{
+    /// <summary>
+    /// Class holds a cache of C# classes and their properties. The key for the cache is the schema full name.
+    /// </summary>
+    public class ClassCache
+    {
+        private ConcurrentDictionary<string, DotnetClass> _nameClassMap = new ConcurrentDictionary<string, DotnetClass>();
+        private static ConcurrentBag<IAvroFieldConverter> _defaultConverters = new ConcurrentBag<IAvroFieldConverter>();
+        private void AddClassNameMapItem(RecordSchema schema, Type dotnetClass)
+        {
+            if (schema != null && GetClass(schema)!=null)
+            {
+                return;
+            }
+
+            if (!dotnetClass.IsClass)
+            {
+                throw new AvroException( $"Type {dotnetClass.Name} is not a class");
+            }
+
+            _nameClassMap.TryAdd(schema.Fullname, new DotnetClass(dotnetClass, schema, this));
+        }
+
+        /// <summary>
+        /// Add a default field converter
+        /// </summary>
+        /// <param name="converter"></param>
+        public static void AddDefaultConverter(IAvroFieldConverter converter)
+        {
+            _defaultConverters.Add(converter);
+        }
+
+        public static void AddDefaultConverter<A,P>(Func<A,Schema, P> from, Func<P,Schema, A> to)
```

Comment:
Add a comment to AddDefaultConverter, and any other externally visible elements. We haven't been good about this in the past in the C# project, but now's a great time to start, especially for new elements. 😄

Reply:
Done.